op-challenger/game/fault/preimages/large.go

```diff
@@ -137,43 +137,23 @@ func (p *LargePreimageUploader) initLargePreimage(ctx context.Context, uuid *big
 // This method **must** be called after calling [initLargePreimage].
 // SAFETY: submits transactions in a [Queue] for latency while preserving submission order.
 func (p *LargePreimageUploader) addLargePreimageData(ctx context.Context, uuid *big.Int, chunks []keccakTypes.InputData) error {
-	queue := txmgr.NewQueue[int](ctx, p.txMgr, 10)
-	receiptChs := make([]chan txmgr.TxReceipt[int], len(chunks))
+	txs := make([]txmgr.TxCandidate, len(chunks))
 	blocksProcessed := int64(0)
 	for i, chunk := range chunks {
 		tx, err := p.contract.AddLeaves(uuid, big.NewInt(blocksProcessed), chunk.Input, chunk.Commitments, chunk.Finalize)
 		if err != nil {
 			return fmt.Errorf("failed to create pre-image oracle tx: %w", err)
 		}
 		blocksProcessed += int64(len(chunk.Input) / keccakTypes.BlockSize)
-		receiptChs[i] = make(chan txmgr.TxReceipt[int], 1)
-		queue.Send(i, tx, receiptChs[i])
+		txs[i] = tx
 	}
-	for _, receiptCh := range receiptChs {
-		receipt := <-receiptCh
-		if receipt.Err != nil {
-			return receipt.Err
-		}
-		if receipt.Receipt.Status == ethtypes.ReceiptStatusFailed {
-			p.log.Error("LargePreimageUploader add leafs tx successfully published but reverted", "tx_hash", receipt.Receipt.TxHash)
-		} else {
-			p.log.Debug("LargePreimageUploader add leafs tx successfully published", "tx_hash", receipt.Receipt.TxHash)
-		}
-	}
-	return nil
+	_, err := p.txSender.SendAndWait(txs...)
+	return err
```

Comment:
Should we change the `LargePreimageUploader`'s `sendTxAndWait` method to accept variadic tx candidate args and then just pass in there with an error return? The `sendTxAndWait` method could then also log any receipt data if we want.

Reply:
Given `TxSender` is handling the logging, I inlined a couple of these `sendTxAndWait` functions which were just stripping the returned receipts.  In `responder` it's still useful to do that as the error is returned directly so it's neater to have a function that only returns an error.  For this and `direct` it's easy to ignore the returned receipts inline so don't need the method.